tests/core/pipeline/test_pipeline_manager.py

```diff
@@ -386,14 +391,37 @@ def test_pipeline_notification_subscribe(mocker):
 
     # test pipeline unsubscribe notification
     # test subscribe notification only on new job
-    _PipelineManager._unsubscribe(notify_1, pipeline)
+    _PipelineManager._unsubscribe(callback=notify_1, pipeline=pipeline)
     _PipelineManager._subscribe(callback=notify_2, pipeline=pipeline)
     _PipelineManager._submit(pipeline.id)
 
     notify_1.assert_not_called()
     notify_2.assert_called_3_times()
 
 
+def test_pipeline_notification_subscribe_multi_param():
+    Config.configure_job_executions(mode=JobConfig._DEVELOPMENT_MODE)
+    _Scheduler._update_job_config()
+
+    pipeline_config = Config.configure_pipeline(
+        "by_6",
+        [
+            Config.configure_task(
+                "mult_by_two",
+                mult_by_two,
+                [Config.configure_data_node("foo", "in_memory", Scope.PIPELINE, default_data=1)],
+                Config.configure_data_node("bar", "in_memory", Scope.PIPELINE, default_data=0),
+            )
+        ],
+    )
+
+    pipeline = _PipelineManager._get_or_create(pipeline_config)
+
+    # test pipeline subscribe notification
+    _PipelineManager._subscribe(callback=notify_multi_param, params=["foobar", 123, 1.2], pipeline=pipeline)
+    _PipelineManager._submit(pipeline.id)
```

Comment:
How can we be sure that the callbacks are well called? We should add a state in the callback to assert it has been called.